extensions/applicationinsights-react-js/src/useTrackEvent.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * ReactPlugin.ts
+ * @copyright Microsoft 2019
+ */
+import { useState, useEffect, useRef } from "react";
+import ReactPlugin from "./ReactPlugin";
+
+export default function useCustomEvent<T>(
+  reactPlugin: ReactPlugin,
+  eventName: string,
+  eventData: T,
+  skipFirstRun = true
```

Comment:
@aaronpowell do you mind explaining why this `skipFirstRun` flag is needed here? thanks.

Reply:
@xiao-lix The reason is to mimic more closely the way the non-hook version works. With `useEffect` hooks the effect is triggered on each value update _including_ the initial setting of the value, meaning we would fire off too early and potentially have unwanted events tracked. I explain it more in the post I wrote: https://www.aaron-powell.com/posts/2019-11-19-combining-react-hooks-with-appinsights/#ignoring-unwanted-effect-runs